packages/react-kit/src/components/Header/Header.tsx

```diff
@@ -58,6 +58,53 @@ export const Header = ({
 					? 'rk-menu-vertical rk-w-full'
 					: 'rk-hidden rk-flex-wrap lg:rk-flex',
 			),
+			overlay: clsx(
+				'rk-w-full',
+				isOpen && windowWidth <= MAX_MOBILE_NAV_WIDTH
+					? 'rk-absolute rk-bottom-0 rk-left-0 rk-z-10 rk-h-[calc(100%_-_72px)] rk-flex-col rk-overflow-y-auto rk-overflow-x-hidden rk-px-4 rk-pb-4 rk-pt-8 sm:rk-px-6 lg:rk-px-12'
+					: 'rk-hidden rk-flex-wrap lg:rk-flex',
+			),
+		};
+
+		const PrimaryNav = () => {
+			if (isNavTuple(mainNavItems)) {
```

Comment:
This does a nice job of allowing both the 'just works' and 'full control' use cases. Likely to be an approach we can use elsewhere. Thanks!